contracts/infrastructure/ArgentWalletDetector.sol

```diff
@@ -0,0 +1,126 @@
+// Copyright (C) 2018  Argent Labs Ltd. <https://argent.xyz>
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.6.12;
+import "./base/Owned.sol";
+import "../wallet/IWallet.sol";
+
+/**
+ * @title ArgentWalletDetector
+ * @notice Simple contract to detect if a given address represents an Argent wallet.
+ * The `isArgentWallet` method returns true if the codehash matches one of the deployed Proxy
+ * and if the target implementation matches one of the deployed BaseWallet.
+ * Only the owner of the contract can add code hash and implementations.
+ * @author Julien Niset - <julien@argent.xyz>
+ */
+contract ArgentWalletDetector is Owned {
+	
+	// The accepeted code hash
+	bytes32[] private codes;
+	// The accepted implementations
+	address[] private implementations;
+	// mapping to efficiently check if a code is accepeted
+    mapping (bytes32 => Info) public acceptedCodes;
```

Comment:
incorrect indentation?

Reply:
No idea what happened with indentation, there is no problem locally.

Reply:
Should be fixed now.